code-studio/src/js/components/progress/progress_dot.jsx

```diff
@@ -156,49 +166,56 @@ export const ProgressDot = React.createClass({
     const outlineCurrent = this.props.courseOverviewPage && onCurrent;
     const smallDot = !this.props.courseOverviewPage && !onCurrent;
     const showLevelName = level.kind === 'named_level' && this.props.courseOverviewPage;
+    const isPeerReview = level.kind === 'peer_review';
 
     return (
       <a
         key='link'
-        href={level.url}
+        href={level.locked ? undefined : level.url}
         onClick={this.props.saveAnswersBeforeNavigation && dotClicked.bind(null, level.url)}
-        style={[styles.outer, showLevelName && {display: 'table-row'}]}
+        style={[styles.outer, (showLevelName || isPeerReview) && {display: 'table-row'}, level.locked && styles.disabledLevel]}
```

Comment:
Can `showLevelName` be set to `/(named_level|peer_review)/.test(level.kind)  && this.props.courseOverviewPage` to clean up all these `|| isPeerReview`?